src/main/drivers/rcc.c

```diff
@@ -28,67 +28,85 @@ void RCC_ClockCmd(rccPeriphTag_t periphTag, FunctionalState NewState)
 
 #if defined(USE_HAL_DRIVER)
 
-#define __HAL_RCC_CLK_ENABLE(bus, enbit)   do {            \
-        __IO uint32_t tmpreg;                              \
-        SET_BIT(RCC->bus ## ENR, enbit);                   \
-        /* Delay after an RCC peripheral clock enabling */ \
-        tmpreg = READ_BIT(RCC->bus ## ENR, enbit);         \
-        UNUSED(tmpreg);                                    \
+// Note on "suffix" macro parameter:
+// ENR and RSTR naming conventions for buses with multiple registers per bus differs among MCU types.
+// ST decided to use AxBn{L,H}ENR convention for H7 which can be handled with simple "ENR" (or "RSTR") contatenation,
+// while use AxBnENR{1,2} convention for G4 which requires extra "suffix" to be concatenated.
+// Here, we use "suffix" for all MCU types and leave it as empty where not applicable.
+
+#define __HAL_RCC_CLK_ENABLE(bus, suffix, enbit)   do {      \
+        __IO uint32_t tmpreg;                                \
+        SET_BIT(RCC->bus ## ENR ## suffix, enbit);           \
+        /* Delay after an RCC peripheral clock enabling */   \
+        tmpreg = READ_BIT(RCC->bus ## ENR ## suffix, enbit); \
+        UNUSED(tmpreg);                                      \
     } while(0)
 
-#define __HAL_RCC_CLK_DISABLE(bus, enbit) (RCC->bus ## ENR &= ~(enbit))
+#define __HAL_RCC_CLK_DISABLE(bus, suffix, enbit) (RCC->bus ## ENR ## suffix &= ~(enbit))
 
-#define __HAL_RCC_CLK(bus, enbit, newState) \
-    if (newState == ENABLE) {               \
-        __HAL_RCC_CLK_ENABLE(bus, enbit);   \
-    } else {                                \
-        __HAL_RCC_CLK_DISABLE(bus, enbit);  \
+#define __HAL_RCC_CLK(bus, suffix, enbit, newState) \
+    if (newState == ENABLE) {                       \
+        __HAL_RCC_CLK_ENABLE(bus, suffix, enbit);   \
+    } else {                                        \
+        __HAL_RCC_CLK_DISABLE(bus, suffix, enbit);  \
     }
 
     switch (tag) {
     case RCC_AHB1:
-        __HAL_RCC_CLK(AHB1, mask, NewState);
+        __HAL_RCC_CLK(AHB1, /**/, mask, NewState);
         break;
 
     case RCC_AHB2:
-        __HAL_RCC_CLK(AHB2, mask, NewState);
+        __HAL_RCC_CLK(AHB2, /**/, mask, NewState);
         break;
 
-#ifndef STM32H7
+#if !(defined(STM32H7) || defined(STM32G4))
     case RCC_APB1:
-        __HAL_RCC_CLK(APB1, mask, NewState);
+        __HAL_RCC_CLK(APB1, /**/, mask, NewState);
```

Comment:
Not sure we want the empty comment here - it is not needed, and it does not make the macro expansion into a valid C function call. If you want to get a valid C function call syntax you'll have to null check `suffix` inside the macro.

Reply:
Hmm... It's a macro call and (although I've never done this before,) I though it would be better to explicitly indicate the argument being empty here rather than leaving it as non-existent argument.

The line does expand into valid statements:
```
    case RCC_APB11:
        if (NewState == ENABLE) { do { volatile uint32_t tmpreg; ((((RCC_TypeDef *) (((0x40000000UL) + 0x00020000UL) + 0x1000UL))->APB1ENR1) |= (mask)); tmpreg = ((((RCC_TypeDef *) (((0x40000000UL) + 0x00020000UL) + 0x1000UL))->APB1ENR1) & (mask)); (void)tmpreg; } while(0); } else { (((RCC_TypeDef *) (((0x40000000UL) + 0x00020000UL) + 0x1000UL))->APB1ENR1 &= ~(mask)); };
        break;
```

And simple test case
```
#define CONCATTEST(a,b) (a ## b)
CONCATTEST(X, Y)
CONCATTEST(X, /**/)
CONCATTEST(/**/, Y)
```
expands into
```
#define CONCATTEST(a,b) (a ## b)
(XY)
(X)
(Y)
```

Thoughts?

Reply:
It does expand into a valid statement, but so does `__HAL_RCC_CLK(APB1, , mask, NewState)`. 

Neither of these are valid C function calls, so my preference would be for the simpler version.

Reply:
Are you suggesting a dummy argument such as
```
#define NOSUFFIX // Empty
__HAL_RCC_CLK(APB1, NOSUFFIX , mask, NewState);
``` 
or just `__HAL_RCC_CLK(APB1, , mask, NewState)`?

Reply:
The `NOSUFFIX` version is neat.